### PR TITLE
Report-month endpoint

### PR DESCRIPTION
**Pull request by carlosper88**
This PR adds the feature requested here (https://horizenlabs.atlassian.net/browse/SCHAINS-219)
It receives two query parameters, from and to, being to optional, so in case it's requested only with from, the endopoint will return all of the months from the from months until today
For a request /report-month?from=202109&to=202110 It returns a json structure like this:
```
[
   {
      "month": 202109,
      "amount": 12345
   },
   {
      "month": 202110,
      "amount": 23421
   },
]
```

### CODE REVIEW

File: src/provider/AuditchainStateProvider.ts [hunk 1]
```diff
@@ -11,4 +14,44 @@ export class AuditchainStateProvider extends SCStateProvider {
...
+
+    async getBoxesInRange(params: ScSCP.GetBoxesInRangeParams) {
+        const fromYear = Number(params.from.substring(0, 4))
+        const fromMonth = Number(params.from.substring(4, 6))
+        const fromDate = new Date(fromYear, fromMonth-1)
+        let filterJoin: any = { 'transaction.blockTime': { $gt: fromDate } }
+        if (params.to) {
+            const toYear = Number(params.to.substring(0, 4))
+            const toMonth = Number(params.to.substring(4, 6))
+            const toDate = new Date(toYear, toMonth-1)
```
Comment (otoumas) [inline src/provider/AuditchainStateProvider.ts:26]: we could have a method for these three lines, to avoid duplication

**Review by otoumas — APPROVED** [review]
LGTM See comments

File: src/provider/AuditchainStateProvider.ts (hunk 1 above)
Comment (carlosper88) [reply src/provider/AuditchainStateProvider.ts]: added!

**Review by otoumas — APPROVED** [review]
LGTM

**Review by paolocappelletti — APPROVED** [review]

**Pull request merged**